packages/protocol/contracts/partners/lido/LidoL1Bridge.sol

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import { Bridge } from "../../bridge/Bridge.sol";
+import { IBridge } from "../../bridge/IBridge.sol";
+import { ILidoL1Bridge } from "./ILidoL1Bridge.sol";
+import { ILidoL2Bridge } from "./ILidoL2Bridge.sol";
+import { BridgeableTokens } from "../../thirdparty/lido/BridgeableTokens.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title LidoL1Bridge
+ * @dev Implementation of the Lido L1 Bridge, extending the ILidoL1Bridge interface and
+ * BridgeableTokens contract
+ */
+contract LidoL1Bridge is ILidoL1Bridge, BridgeableTokens {
+    using SafeERC20 for IERC20;
+
+    IBridge bridge;
+    uint32 destChainId;
+    address public lidoL2Bridge;
+    mapping(bytes32 => bool) failedMsgHashes;
+
+    error Lido_notSelf();
+    error Lido_notL2Bridge();
+    error Lido_messageTampered();
+    error Lido_messageNotFailed();
+    error Lido_messageProcessingFailed();
+    error Lido_incorrectFundsTransferred();
+    error Lido_failedMsgAlreadyProcessed();
+
+    /**
+     * @dev Modifier to restrict function access to only the contract itself
+     */
+    modifier onlySelf() {
+        if (msg.sender != address(this)) revert Lido_notSelf();
+        _;
+    }
+
+    /**
+     * @dev Modifier to restrict function access to only the L2 bridge
+     * @param bridge_ The address to be checked against the Lido L2 bridge address
+     */
+    modifier onlyL2Bridge(address bridge_) {
+        if (bridge_ != lidoL2Bridge) revert Lido_notL2Bridge();
+        _;
+    }
+
+    /**
+     * @dev Modifier to ensure correct transfer of L1 tokens before and after a function call.
+     * @param isTo Boolean indicating whether tokens are being transferred to (`true`) or from
+     * (`false`) the contract.
+     * @param amount_ The amount of tokens being transferred.
+     */
+    modifier transferL1Tokens(bool isTo, uint256 amount_) {
```

Comment:
if the token is any arbitrary ERC20 token, then we need this diff-check, if we know it's a specific token implementaiton, we don't.

Is Lido token on L1 a regular ERC20 token?

Reply:
Yes regular erc20. Will remove check